### PR TITLE
ICU-20197 ICU4C Multithreaded tests cleanup

### PR DESCRIPTION
**Pull request by aheninger**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20197
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/tsmthred.cpp [hunk 1]
```diff
@@ -451,8 +439,8 @@ const ThreadSafeFormatSharedData *gSharedData = NULL;
 
 ThreadSafeFormatSharedData::ThreadSafeFormatSharedData(UErrorCode &status) {
     fFormat.adoptInstead(NumberFormat::createCurrencyInstance(Locale::getUS(), status));
-    static const UChar kYDD[] = { 0x59, 0x44, 0x44, 0x00 };
-    static const UChar kBBD[] = { 0x42, 0x42, 0x44, 0x00 };
+    static const UChar kYDD[] = { u'Y', u'D', u'D', 0x00 };
```
Comment (sffc) [inline icu4c/source/test/intltest/tsmthred.cpp]: static const UChar* kYDD = u"YDD";

File: .travis.yml [hunk 2]
```diff
@@ -106,7 +106,7 @@ matrix:
     - name: "c: linux tsan"
       language: cpp
       env:
-          - INTLTEST_OPTS="utility/MultithreadTest rbbi/RBBIMonkeyTest format/CalendarLimitTest"
+          - INTLTEST_OPTS="utility/MultithreadTest"
```
Comment (sffc) [inline .travis.yml]: Did you move the old tests? I don't see that.  If you do move them, how about adding a top-level "thread" category parallel to utility.

File: icu4c/source/test/intltest/tsmthred.cpp [hunk 3]
```diff
@@ -777,7 +763,7 @@ void MultithreadTest::TestThreadedIntl()
     UErrorCode threadSafeErr = U_ZERO_ERROR;
 
     ThreadSafeFormatSharedData sharedData(threadSafeErr);
-    assertSuccess("initializing ThreadSafeFormat", threadSafeErr, TRUE);
+    assertSuccess(WHERE, threadSafeErr, TRUE);
```
Comment (sffc) [inline icu4c/source/test/intltest/tsmthred.cpp]: what is `WHERE`?

File: icu4c/source/test/intltest/tsmthred.cpp (hunk 1 above)
Comment (aheninger) [reply icu4c/source/test/intltest/tsmthred.cpp]: Yes, done.

File: icu4c/source/test/intltest/tsmthred.cpp (hunk 3 above)
Comment (aheninger) [reply icu4c/source/test/intltest/tsmthred.cpp]: It's a macro in the test framework for the source file and line number. Makes a nice message for test failures.

File: .travis.yml (hunk 2 above)
Comment (aheninger) [reply .travis.yml]: The tests outside of the MultithreadTest directory are threaded primarily to reduce the elapsed runtime of long-running functional tests for some service or another. They run too long to conveniently run under the thread sanitizer, which is really slow. 

Bottom line, I think they're fine where they are, with the service they're testing.

**Pull request merged**